akka-remote/src/main/scala/akka/remote/serialization/DaemonMsgCreateSerializer.scala

```diff
@@ -186,26 +185,23 @@ private[akka] final class DaemonMsgCreateSerializer(val system: ExtendedActorSys
 
     // this trixery is to retain backwards wire compatibility while at the same time
     // allowing for usage of serializers with string manifests
-    var hasManifest = false
+    val hasManifest = serializer.includeManifest
     val manifest = serializer match {
```

Comment:
replacing this with `Serializers.manifestFor` doesn't affect tests even on scala 2.12, but I still think this can break compatibility with older nodes. WDYT?

Reply:
the serialization here is messy, but don't change it please